Shovit_Kandel/Lion_Dog_Assignment/dog.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'mammal'
+class Dog < Mammal
+	def pet
+		@health += 5
+		self
+	end
+	def walk
+		@health -= 1
+		self
+	end
+	def run
+		@health -= 10
+		self
+	end
+end
+Dog.new.walk.walk.walk.run.run.pet.display_health
+
+require_relative 'mammal'
+class Lion < Mammal
+	def initialize
+		@health = 170
+		self
+	end
+	def fly
+		@health -=10
+		self
+	end
+	def attack_town
+		@health -=50
+		self
+	end
+	def eat_humans
+		@health += 20
+		self
+	end
+	def display_health
+		Mammal.instance_method(:display_health).bind(self).call
```

Comment:
Interesting way to do this, I haven't seen it done this way before.  Another way to do it though is to just use 'super' instead of that chain of methods...when a method calls 'super' it inherets directly from the same named method in the parent class, so it would look like this:
def display_health
    super
    puts "This is a dragon"
    self
end